File: solutions/go/sum-of-multiples/1/sum_of_multiples.go
package summultiples

func SumMultiples(limit int, divisors ...int) (sum int) {
	for i := 1; i < limit; i++ {
		for _, v := range divisors {
			if v != 0 && i%v == 0 {
				sum += i
				break
			}
		}
	}

	return
}

File: solutions/typescript/sum-of-multiples/1/sum-of-multiples.ts
export function sum(factors: number[], limit: number): number {
  let sum = 0

  for (let i = 1; i < limit; i++) {
    for (const f of factors) {
      if (f !== 0 && i % f === 0) {
        sum += i
        break
      }
    }
  }

  return sum
}

// export function sum(factors: number[], limit: number): number {
//   return Array.from({ length: limit - 1 }, (_, i) => i + 1)
//     .filter(num => factors.some(f => f !== 0 && num % f === 0))
//     .reduce((acc, num) => acc + num, 0)
// }
